### PR TITLE
Add coverage to http classes

### PR DESCRIPTION
**Pull request by Christoffer-Cortes**
### Applicable Issues
Coverage is very low atm in this repo.

### Description of the Change
Increased coverage of http classes to 97%.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: @Christoffer-Cortes

### CODE REVIEW

File: src/main/java/com/ericsson/eiffelcommons/http/HttpRequest.java [hunk 1]
```diff
@@ -331,10 +331,12 @@ private URIBuilder addParametersToURIBuilder(URIBuilder builder) {
      * @throws URISyntaxException
      */
     private URIBuilder createURIBuilder() throws URISyntaxException {
-        if (endpoint.startsWith("/")) {
+        if (endpoint != null && !endpoint.isEmpty() && endpoint.startsWith("/")) {
```
Comment (a-breid-ericsson) [inline src/main/java/com/ericsson/eiffelcommons/http/HttpRequest.java]: Change to StrungUtils.isEmpty

File: pom.xml [hunk 2]
```diff
@@ -49,6 +57,18 @@
             <version>20180130</version>
         </dependency>
 
+        <dependency>
```
Comment (a-breid-ericsson) [inline pom.xml]: Do not think we have or should have logging in this repository. If something fails we should throw exceptions with messages to the main application may log them.

File: src/main/java/com/ericsson/eiffelcommons/http/HttpRequest.java (hunk 1 above)
Comment (Christoffer-Cortes) [reply src/main/java/com/ericsson/eiffelcommons/http/HttpRequest.java]: I had that earlier and travis couldn't find the package. I can try later again.

File: pom.xml (hunk 2 above)
Comment (Christoffer-Cortes) [reply pom.xml]: Some dependency I added started giving out warning of logging binders so I added it to make it stop. Maybe the warnings were there before, I'm not sure.

Comment (Christoffer-Cortes) [reply pom.xml]: Took a look at this again and it was only  the executor tests giving this warning. I decided to ignore this warning and remove the logger.

File: src/main/java/com/ericsson/eiffelcommons/http/HttpRequest.java (hunk 1 above)
Comment (Christoffer-Cortes) [reply src/main/java/com/ericsson/eiffelcommons/http/HttpRequest.java]: Travis had some temporary problems it seems. StringUtils now seems to work.

**Pull request merged**